tools/hashes_collection.sh

```
@@ -1,18 +1,60 @@
-# explanation:
-# 1. we select logical path and data object size for each data object within the collection tree
-# 2. we remove the prefix (collection name) so that the logical name starts within the collection (needed for comparison)
-# 3. we sort the pathnames, using a locale that ensures all characters have a different sorting order
-# bash commands:
-
-# Check if COLLECTION argument is provided
-if [ -z "$1" ]; then
-    echo "Error: Collection path must be provided as an argument." >&2
-    echo "Usage: $0 <collection-path>" >&2
-    exit 1
+
+#set of shell commands to hash the set of data objects and hierarchy in a collection
+#To compare the content of 2 collections, update the shell variable COLLECTION and rerun the query line
+#explanation:
+#1. we select logical path and data object size for each data object within the collection tree
+#2. we remove the prefix (collection name) so that the logical name starts within the collection (needed for comparison)
+#3. we sort the pathnames, using a locale that ensures all characters have a different sorting order
+
+#bash commands:
+
+DETAILED=0
+DEBUG=0
+
+# Parse arguments
+for arg in "$@"; do
+    case "$arg" in
+        --detailed) DETAILED=1 ;;
+        --debug) DEBUG=1 ;;
+        *) if [[ -z "$COLLECTION" ]]; then COLLECTION="${arg%/}"; fi ;;
+    esac
+done
+
+if [ -z "$COLLECTION" ]; then
+   echo "Usage: bash $0 <collectionName> [--detailed] [--debug]"
+   exit 1
 fi
 
-COLLECTION="${1%/}/"
-iquest --no-page "%s/%s %s" "select COLL_NAME,DATA_NAME,DATA_SIZE where COLL_NAME like '${COLLECTION}%'"|cut -c$((${#COLLECTION}+1))-|LC_ALL=C sort|sha256sum
+DIR="/tmp"
+export TEMPFILE=${DIR}/check_objects.tmp
+export TEMPFOLDER=${DIR}/check_collections.tmp
+export TEMPSORTED=${DIR}/check_sorted.tmp
+
+iquest --no-page "%s/%s %s" "select COLL_NAME,DATA_NAME,DATA_SIZE where COLL_NAME like '${COLLECTION}/%'" | grep -v '^CAT_NO_ROWS_FOUND' | cut -c$((${#COLLECTION}+1))- >${TEMPFILE}
```

This query, as well as the one below, also compares non-good (e.g. `stale` or `intermediate`) replicas if they have different sizes. That's probably not what we want as default behaviour. I suggest adding a `DATA_REPL_STATUS` condition to these queries. If we decide to keep it out of scope of the present ticket,  we need to at least document this as a limitation of the current version of the script.

It would also be good to check the return code of the `iquest` queries here, so that we can return an error message if something goes wrong with the queries, rather than a checksum that is not based on the real contents of the collection. We can't use `set -o pipefail` here, because `CAT_NO_ROWS_FOUND` with return code `1` is a valid result, but we can check for codes other than 0/1 using `PIPESTATUS`

---

Changed as suggested.